crates/ruff_linter/src/rules/flake8_pyi/rules/type_comment_in_stub.rs

```diff
@@ -41,12 +42,13 @@ pub(crate) fn type_comment_in_stub(
     diagnostics: &mut Vec<OldDiagnostic>,
     locator: &Locator,
     comment_ranges: &CommentRanges,
+    source_file: &SourceFile,
```

Comment:
Nit: We could consider introducing a `DiagnosticsCollector` or similar which is a thin wrapper around a `Vec<OldDIagnostic>` and ` SourceFile` with a `report_diagnostic` method. This would avoid introducing more arguments and would allow us to use the same pattern in more places.